src/GitLink/Linker.cs

```diff
@@ -33,74 +33,80 @@ public static void Link(string pdbPath, LinkOptions options = default(LinkOption
         {
             Argument.IsNotNullOrEmpty(() => pdbPath);
 
-            var pdb = new PdbFile(pdbPath);
-            var filesAndChecksums = pdb.GetFiles();
-            var sourceFiles = filesAndChecksums.Select(f => f.Item1);
-
-            string repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
-            using (var repository = new Repository(repositoryDirectory))
+            var projectSrcSrvFile = pdbPath + ".srcsrv";
+            string repositoryDirectory;
+            using (var pdb = new PdbFile(pdbPath))
             {
-                var providerManager = new Providers.ProviderManager();
-                Providers.IProvider provider;
-                if (options.GitRemoteUrl == null)
-                {
-                    var candidateProviders = from remote in repository.Network.Remotes
-                                             let p = providerManager.GetProvider(remote.Url)
-                                             where p != null
-                                             select p;
-                    provider = candidateProviders.First();
-                }
-                else
-                {
-                    provider = providerManager.GetProvider(options.GitRemoteUrl.AbsoluteUri);
-                }
-
-                var projectSrcSrvFile = pdbPath + ".srcsrv";
+                var sourceFiles = pdb.GetFiles().Select(f => f.Item1).ToList();
 
-                if (!options.SkipVerify)
+                repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
+                using (var repository = new Repository(repositoryDirectory))
                 {
-                    Log.Info("Verifying pdb file");
+                    var repoSourceFiles = sourceFiles.ToDictionary(e => e, repository.GetRepoNormalizedPath);
 
-                    var missingFiles = pdb.FindMissingOrChangedSourceFiles();
-                    foreach (var missingFile in missingFiles)
+                    var providerManager = new Providers.ProviderManager();
+                    Providers.IProvider provider;
+                    if (options.GitRemoteUrl == null)
                     {
-                        Log.Warning($"File \"{missingFile}\" missing or changed since the PDB was compiled.");
+                        var candidateProviders = from remote in repository.Network.Remotes
+                                                 let p = providerManager.GetProvider(remote.Url)
+                                                 where p != null
+                                                 select p;
+                        provider = candidateProviders.First();
```

Comment:
Do we want to crash here or log an error so the user actually knows what is crashing this? I prefer the latter.